v0/destinations/wootric/util.js

```diff
@@ -0,0 +1,79 @@
+const qs = require("qs");
+const { httpGET, httpPOST } = require("../../../adapters/network");
+const {
+  BASE_ENDPOINT,
+  VERSION,
+  ACCESS_TOKEN_CACHE_TTL_SECONDS
+} = require("./config");
+const { CustomError } = require("../../util");
+const axios = require("axios");
+const Cache = require("../../util/cache");
+
+const ACCESS_TOKEN_CACHE = new Cache(ACCESS_TOKEN_CACHE_TTL_SECONDS);
+
+const getAccessToken = async destination => {
+  const { username, password, accountToken } = destination.Config;
+  const accessTokenKey = destination.ID;
+
+  return ACCESS_TOKEN_CACHE.get(accessTokenKey, async () => {
+    const request = {
+      header: {
+        "Content-Type": "application/x-www-form-urlencoded",
+        Accept: "application/json"
+      },
+      url: `${BASE_ENDPOINT}/oauth/token?account_token=${accountToken}`,
+      data: qs.stringify({
+        grant_type: "password",
+        username,
+        password
+      }),
+      method: "POST"
+    };
+    const wootricAuthResponse = await httpPOST(
+      request.url,
+      request.data,
+      request.header
+    );
+    // If the request fails, throwing error.
+    if (wootricAuthResponse.success === false) {
+      throw new CustomError(
+        `[Wootric]:: access token could not be generated due to ${wootricAuthResponse.response.data.error}`,
+        400
+      );
+    }
+    return wootricAuthResponse.response?.data?.access_token;
+  });
+};
+
+const retrieveUserId = async (userId, destination) => {
+  try {
+    const accessToken = await getAccessToken(destination);
+    if (!accessToken) {
+      throw new CustomError(`[Wootric]:: access token is not available`, 400);
+    }
+
+    const endpoint = `${BASE_ENDPOINT}/${VERSION}/end_users/${userId}?lookup_by_external_id=true`;
+    const requestOptions = {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${accessToken}`
+      }
+    };
+
+    const userResponse = await httpGET(endpoint, requestOptions);
+    // If the request fails, throwing error.
+    if (userResponse.success === false) {
+      throw new CustomError(
+        `[Wootric]:: Unable to retrieve userid due to ${userResponse.response
+          ?.data?.error ?? "unkown error"}`,
```

Comment:
please check the `??` here. IMO it will be `||`. Also, the error structure as I see is: 

```
{
  "success": false,
  "response": {
    "message": "Request failed with status code 404",
    "name": "Error",
    "stack": "Error: Request failed with status code 404\n    at createError (/Users/shrouti/workspace/rudder-transformer/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/Users/shrouti/workspace/rudder-transformer/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/Users/shrouti/workspace/rudder-transformer/node_modules/axios/lib/adapters/http.js:322:11)\n    at IncomingMessage.emit (events.js:388:22)\n    at endReadableNT (internal/streams/readable.js:1336:12)\n    at processTicksAndRejections (internal/process/task_queues.js:82:21)",
    "config": {
      "transitional": {
        "silentJSONParsing": true,
        "forcedJSONParsing": true,
        "clarifyTimeoutError": false
      },
      "transformRequest": [
        null
      ],
      "transformResponse": [
        null
      ],
      "timeout": 0,
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
      "maxContentLength": -1,
      "maxBodyLength": -1,
      "headers": {
        "Accept": "application/json, text/plain, */*",
        "Content-Type": "application/x-www-form-urlencoded",
        "User-Agent": "axios/0.26.1",
        "Content-Length": 87
      },
      "Content-Type": "application/x-www-form-urlencoded",
      "Accept": "application/json",
      "method": "post",
      "url": "https://api.wootric.com/oauth/token?account_token=13da035041f572e8eff60743662f6b87e5eaebd8eafd7cab3a437df9a200a6f7",
      "data": "grant_type=password&username=gauravudia%40rudderstack.com&password=n5f5XH4Xxp%26%3F3rEY"
    },
    "status": 404
  }
}
```
not finding data fields in the error message. can you recheck that as well